src/main/java/org/apache/maven/plugin/compiler/TestCompilerMojo.java

```diff
@@ -380,28 +384,300 @@ else if ( Double.valueOf( getTarget() ) < Double.valueOf( MODULE_INFO_TARGET ) )
+        {
+            getLog().debug( "patchModule test paths:" );
+            for ( String moduleTestPath : listModuleTestPaths ) 
+            {
+                getLog().debug( "  " + moduleTestPath );
+            }
+            
+        }
+        
+        // Get modularized dependencies resolved before
+        for ( Entry<File, ModuleNameSource> pathElt : result.getModulepathElements().entrySet() ) 
+        {
+            
+            File path = pathElt.getKey();
+            ModuleNameSource moduleNameSource = pathElt.getValue();
+            
+            // Get module name
+            JavaModuleDescriptor moduleDesc = result.getPathElements().get( path );
+            String moduleName = ( moduleDesc != null ) ? moduleDesc.name() : null;
+            
+            if ( 
+                  // Is it a modularized compile elements?
+                  ( path != null ) && ( moduleName != null )
+                  &&
+                  // Not an auto-module
+                  !moduleDesc.isAutomatic()
+                  &&
+                  // Is it different from main module name
+                  !mainModuleName.equals( moduleName )
+            ) 
+            {
+            
+                // Add --add-reads <moduleName>=ALL-UNNAMED
+                compilerArgs.add( "--add-reads" );
+                StringBuilder sbAddReads = new StringBuilder();
+                sbAddReads.append( moduleName ).append( "=ALL-UNNAMED" );
+                compilerArgs.add( sbAddReads.toString() );
+                
+                // Add compile classpath if needed
+                if ( !listModuleTestPaths.isEmpty() )
+                {
+                    // Yes, add it as patch module
+                    List<String> listPath = patchModules.get( moduleName );
+                    // Make sure it is initialized
+                    if ( listPath == null ) 
+                    {
+                        listPath = new ArrayList<>();
+                        patchModules.put( moduleName, listPath );
+                    } 
+                    
+                    // Add test compile path but not main module target
+                    listPath.addAll( listModuleTestPaths );
+                }
+                
+            }
+            
+        }
+
+        if ( pathElements == null ) 
+        {
+            pathElements = new LinkedHashMap<>( result.getPathElements().size() );
+            for ( Entry<File, JavaModuleDescriptor> entry : result.getPathElements().entrySet() ) 
+            {
+                pathElements.put( entry.getKey().getAbsolutePath(), entry.getValue() );
+            }
+        }
+    }
+
+    /**
+     * Get module test path element from module path element
+     * @param modulePathElt
+     * @return
+     */
+    private File getModuleTestPathElt( File modulePathElt )
+    {
+
+        // Get test path from reactor projects
+        File result = getModuleTestPathEltFromReactorProjects( modulePathElt );
```

Comment:
Can you change this to `return getModuleTestPathEltFromReactorProjects( modulePathElt );`

I can't think of a reason why we need the rest of the code in this method. It would mean something went wrong somewhere else.